### PR TITLE
Issue 4: Encapsulate root key in a PKCS12 container

### PR DESCRIPTION
**Pull request by ericeberry**

### CODE REVIEW

**Review by JRHewitt-CC — DISMISSED** [review]

File: reference/src/util/include/common.h [hunk 1]
```diff
@@ -40,11 +39,20 @@
...
+#define DH_4096_BYTE_LENGTH 512UL
 #define DH_MAX_MOD_SIZE 512
 #define RSA_PKCS1_PADDING_SIZE 11
 #define RSA_OAEP_PADDING_SIZE 42
 #define CHACHA20_NONCE_LENGTH 12
 #define CHACHA20_COUNTER_LENGTH 4
 #define CHACHA20_TAG_LENGTH 16
+#define MAX_NUM_SLOTS 256
+
+#define DEFAULT_ROOT_KEYSTORE_PASSWORD "@UJ2@&69&3vsW2@8lg#r4#AU2&Zzv5"
```
Comment (mhabrat) [inline reference/src/util/include/common.h:56]: Recommend adding comment around the intent of this password and why there are no security concerns with embedding it in source files.

File: reference/src/util/include/pkcs12.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#include <openssl/pkcs12.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Loads a secret key from a PKCS12 file identified at the ROOT_KEYSTORE environment variable. The password to the
+ * keystore is in the ROOT_KEYSTORE_PASSWORD environment variable. If neither is defined, it looks for the file
+ * root_keystore.p12 in the working directory with a default password.
```
Comment (mhabrat) [inline reference/src/util/include/pkcs12.h:37]: Couldn't hurt to add explanation on why this is acceptable and not a security risk.

**Review by mhabrat — COMMENTED** [review]
Is there a reason to keep the PKCS 12 keystore at the root of reference rather than an appropriate sub-folder?

File: reference/src/util/include/common.h (hunk 1 above)
Comment (ericeberry) [reply reference/src/util/include/common.h:56]: Added to readme.

File: reference/src/util/include/pkcs12.h (hunk 2 above)
Comment (ericeberry) [reply reference/src/util/include/pkcs12.h:37]: Added to readme

**Review by mhabrat — APPROVED** [review]

**Pull request merged**